RevenueCatUI/Templates/Components/PackageGroup/PackageComponentViewModel.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PackageComponentViewModel.swift
+//
+//  Created by Josh Holtz on 9/27/24.
+
+import Foundation
+import RevenueCat
+
+// swiftlint:disable missing_docs
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+public class PackageComponentViewModel {
```

Comment:
You might have planned for this so FFTI me but I was thinking we are currently creating view models before we verify the package id is valid and the products are found in the store. In case the package/product isn't found, we would probably need to remove this view model and view from the tree somehow... 

I think we might want to have a processing step before creating these view models/views to make sure they are valid before adding them to the view tree.

Again, not a blocker for this PR, just a thought.

Reply:
Oh I see in the other PRs that you are failing if the package can't be found... personally I feel we should just log and continue displaying the paywall (as long as there is at least a valid package). But lmk what you think!